src/splinterdb.c

```diff
@@ -338,23 +397,33 @@ splinterdb_create_or_open(const splinterdb_config *kvs_cfg,      // IN
    task_system_destroy(kvs->heap_id, &kvs->task_sys);
 deinit_iohandle:
    io_handle_deinit(&kvs->io_handle);
+io_handle_init_failed:
+   heap_handle = kvs->heap_handle;
 deinit_kvhandle:
-   platform_free(kvs_cfg->heap_id, kvs);
+   // Depending on the place where a configuration / setup error lead
+   // us to here via a 'goto', heap_id handle, if in use, may be in a
+   // different place. Use one carefully, to avoid ASAN-errors.
+   platform_free((kvs->heap_id ? kvs->heap_id : kvs_cfg->heap_id), kvs);
+   if (we_created_heap) {
+      platform_heap_destroy(&heap_handle);
```

Comment:
This whole error-handling backout block is touchy. I have finessed it "just-right" for all the existing test cases that go thru this 'wrong-param-fail-error-check' test scenarios.

Reply:
Resolved. Reworked interfaces to re-introduce const splinterdb_config *kvs_cfg.